lib/linalg/BasisReader.h

```diff
@@ -276,6 +276,26 @@ class BasisReader {
         double time,
         double ef);
 
+    /**
+     *
+     * @brief Returns the dimension of the system on this processor.
+     *
+     * @return The dimension of the system on this processor.
+     */
+    int
+    getDim(
+        double time);
+
+    /**
+     *
+     * @brief Returns the number of samples (columns) in file.
+     *
+     * @return The number of samples in file.
+     */
+    int
+    getNSamples(
```

Comment:
For consistency with other functions that use `Num`, could we rename this `getNumSamples`? It's hard to change names later, since it can break backwards compatibility.